src/Stack/Build/Execute.hs

```diff
@@ -661,6 +655,35 @@ executePlan' installedMap0 targets plan ee@ExecuteEnv {..} = do
                   $ Map.elems
                   $ planUnregisterLocal plan
 
+unregisterPackages ::
+       (HasProcessContext env, HasLogFunc env)
+    => WhichCompiler
+    -> ActualCompiler
+    -> Path Abs Dir
+    -> [(GhcPkgId, (PackageIdentifier, Text))]
+    -> RIO env ()
+unregisterPackages wc ac localDB ids = do
+    let logReason ident reason =
+            logInfo $
+            fromString (packageIdentifierString ident) <> ": unregistering" <>
+            if T.null reason
+                then ""
+                else " (" <> RIO.display reason <> ")"
+    case ac of
+        ACGhc v | v >= mkVersion [8, 0, 1] -> do
+                let batchSize = 500
```

Comment:
Add a comment on the choice of `500`? I assume max command line argument length?

If yes, you may want to adjust this depending on the OS, e.g. IIRC Windows can only do 8 KB long invocations, and on Linux it's `getconf ARG_MAX`, which returns to `2097152` on my Ubuntu, and according to my strace is computed as

```c
getrlimit(RLIMIT_STACK, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
```

Reply:
Yeah, good catch, Windows was my main thought, I remember older its versions supporting only 255 characters in paths...

Reply:
I've added a comment and switched to 100 when using Windows

Reply:
Nice